### DISCUSSION TITLE
detecting the first and last run in k-way peeksort

### DISCUSSION THREAD

**Discussion by roarin-roran** [roarin-roran/Sorting#65] [General]
I'm currently detecting the first and last run for k-way peeksort, because it allows me to ensure that the code correctly handles runs at the beginning and the end of the first recursive call (full recursion is not yet implemented).

@sebawild should I do this going forward? if not, why not?

**Comment by sebawild** [comment]
I'm not a fan of doing this. Detecting runs eagerly usually bring parts of the array into cache that we don't immediately merge (because of the recursive calls coming first), so in terms of memory transfers, these are expensive, but not effective. Unlike in other cases, I don't think this is a trade-off, my guess is that it is strictly worse to be eager in detecting runs.

**Reply by roarin-roran** [reply]
sure thing, will label it as something to take out when looking at the full version.
